OpenSim/Simulation/Model/Geometry.h

```diff
@@ -180,8 +150,34 @@ class OSIMSIMULATION_API Geometry : public Component {
     /// Methods to support frame as a connection, implement Component interface
     void constructConnectors() override;
 
+protected:
+    /// Map this Geometry into a list of primitives aka SimTK::DecorativeGeometry 
+    /// and return the result in the passed in Array.
+    virtual void createDecorativeGeometry(
+        SimTK::Array_<SimTK::DecorativeGeometry>&) const = 0;
+
 private:
-    
+    /// Compute Transform of this geometry relative to its base frame, utilizing 
+    /// passed in state. Both transform and body_id are set in the passed-in 
+    /// decorations as a side effect.
+    void setDecorativeGeometryTransform(
+        SimTK::Array_<SimTK::DecorativeGeometry>& decorations,
+        const SimTK::State& state) const;
+
+    /// Manage Appearance (how the Geometry is rendered) by applying Appearance 
+    /// from Geometry to DecorativeGeometry.
+    void setDecorativeGeometryAppearance(
+        SimTK::DecorativeGeometry& decoration) const {
+        decoration.setColor(get_Appearance().get_color());
+        decoration.setOpacity(get_Appearance().get_opacity());
+        if (get_Appearance().get_visible())
+            decoration.setRepresentation(
+                (VisualRepresentation)
+                get_Appearance().get_representation());
```

Comment:
Correct, SimTK::DecorativeGeometry has a different set of options that we map to. Representation belongs to SurfaceAppearance since Curves can't have shading/lighting-model, or drawn in wire-frame etc. Good point on texture not handled by the DecorativeGeometry altogether or the API visualizer. We'll have to decide, later, if it will stay on the GUI side (you apply textures once to create pretty picture or movies) or be pushed to API (if/when the API visualizer and GUI converge).